CMakeLists.txt

```diff
@@ -253,23 +252,55 @@ if (CLR_CMAKE_PLATFORM_UNIX)
     message("Detected FreeBSD amd64")
   endif(CLR_CMAKE_PLATFORM_FREEBSD)
 
+  # set the CLANG sanitizer flags for debug build
+  if(UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG)
+    # obtain settings from running enablesanitizers.sh
+    string(FIND "$ENV{DEBUG_SANITIZERS}" "asan" __ASAN_POS)
+    string(FIND "$ENV{DEBUG_SANITIZERS}" "ubsan" __UBSAN_POS)
+    if ((${__ASAN_POS} GREATER -1) OR (${__UBSAN_POS} GREATER -1))
+      # set compiler flags
+      # -fdata-sections -ffunction-sections: each function has own section instead of one per .o file (needed for -Wl and --gc-sections)
+      # -O1: basic level of optimization
```

Comment:
The optimization flag should not be set here. This flags is set in `src\pal\tools\clang-compiler-override.txt` to -O0. Would the sanitizer have problem with it? It is also not clear to me which one would the compiler use if you add the -O1 here.